src/librustc_data_structures/obligation_forest/mod.rs

```diff
@@ -162,38 +204,119 @@ pub struct ObligationForest<O: ForestObligation> {
     ///
     /// [details]: https://github.com/rust-lang/rust/pull/53255#issuecomment-421184780
     error_cache: FxHashMap<ObligationTreeId, FxHashSet<O::CacheKey>>,
+
+    /// Stores which nodes would be unblocked once `O::Variable` is unified
+    stalled_on: FxHashMap<O::Variable, Vec<NodeIndex>>,
+    /// Stores the node indices that are unblocked and should be processed at the next opportunity
+    unblocked: BinaryHeap<Unblocked>,
+    /// Stores nodes which should be processed on the next iteration since the variables they are
+    /// actually blocked on are unknown
+    stalled_on_unknown: Vec<NodeIndex>,
+    /// The offset that this `ObligationForest` has registered. Should be de-registered before
```

Comment:
This is the offset that was registered with the inference table for tracking modifications? Can we make this comment more explicit?